ksql-parser/src/main/java/io/confluent/ksql/parser/tree/Type.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.parser.tree;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public abstract class Type extends Expression {
+
+
+  public enum KsqlType {
```

Comment:
why do we have an enum for `KsqlType` and a class hierarchy extending `Type`? IMO it would be better if they `enum` was for `PrimitiveType`s only and was an internal thing. At the moment we have too many things like Schema, Type, etc that have corresponding `XXXXUtil` classes rather than having objects and using polymorphism. We need to address this